std/assembly/map.ts

```diff
@@ -66,8 +73,42 @@ export class Map<K,V> {
   private entriesOffset: i32 = 0;
   private entriesCount: i32 = 0;
 
-  constructor() {
-    /* nop */
+  constructor(initialEntries: MapInitialEntries<K,V> | null = null) {
+    if (initialEntries) { 
+      if (initialEntries.length >= this.entriesCapacity) this.bucketsMask = initialEntries.length;
```

Comment:
I think the predicate will be always true here, so just set `this.bucketsMask to initial len`
```suggestion
      this.bucketsMask = initialEntries.length;
```